src/named.rs

```diff
@@ -391,3 +402,89 @@ impl<'a, 'b> NamedTempFileOptions<'a, 'b> {
 
     }
 }
+
+/// A path to a named temporary file without an open file handle.
+///
+/// This is useful when the temporary file needs to be used by a child process,
+/// for example.
+///
+/// When dropped, the temporary file is deleted.
+pub struct TempPath(PathBuf);
+
+impl TempPath {
+    /// Close and remove the temporary file.
+    ///
+    /// Use this if you want to detect errors in deleting the file.
+    pub fn close(self) -> io::Result<()> {
+        fs::remove_file(&self.0)
+    }
+
+    /// Persist the temporary file at the target path.
+    ///
+    /// If a file exists at the target path, persist will atomically replace it.
+    /// Upon failure, `self` is returned in the error result.
+    ///
+    /// Note: Temporary files cannot be persisted across filesystems.
+    ///
+    /// *SECURITY WARNING:* Only use this method if you're positive that a
+    /// temporary file cleaner won't have deleted your file. Otherwise, you
+    /// might end up persisting an attacker controlled file.
+    pub fn persist<P: AsRef<Path>>(self, new_path: P) -> Result<(), (TempPath, io::Error)> {
+        match imp::persist(&self.0, new_path.as_ref(), true) {
+            Ok(_) => Ok(()),
```

Comment:
You'll need to keep the drop glue from running here (otherwise, we'll try to delete the file we've just moved).

Reply:
Great catch. `std::mem::forget` should fix that.